### PR TITLE
feat: add parameter for SSH options

### LINKED ISSUE #29: Proposal: Add options for SSH identity file and passphrase to Docker provider
### Terraform Version
<code>&#8805; 0.12.25</code>

### Use-cases
The provider has a feature to deploy a docker container using the SSH protocol
```hcl
provider "docker" {
  host = "ssh://user@remote-host:22"
}
```
However there seems to be no explicit way to provide the path to an SSH identity file (private key).
The only implicit workaround seems to be creating an SSH config file on the machine running Terraform to specify the identity file.

### Proposal
Add two options:
```
identity_file
passphrase
```
Use this information in the SSH systemcall.

### References
Issue in archived repo: https://github.com/hashicorp/terraform-provider-docker/issues/268
Related: https://github.com/kreuzwerker/terraform-provider-docker/issues/18

### PR DESCRIPTION
**Pull request by innovate-invent**
In place of #301 

Resolves #29

### CODE REVIEW

**Comment by innovate-invent** [comment]
@aicarmic I was busy force-pushing my branch and may have killed a commit you pushed

**Comment by innovate-invent** [comment]
@suzuki-shunsuke here you go

File: internal/provider/config.go [hunk 1]
```diff
@@ -118,7 +119,7 @@ func (c *Config) NewClient() (*client.Client, error) {
 	}
 
 	// If there is no cert information, then check for ssh://
-	helper, err := connhelper.GetConnectionHelper(c.Host)
+	helper, err := connhelper.GetConnectionHelperWithSSHOpts(c.Host, c.SSHOpts)
```
Comment (suzuki-shunsuke) [inline internal/provider/config.go:122]: 👍 Even if SSHOpts is nil, there is no problem.
https://github.com/docker/cli/blob/v20.10.12/cli/connhelper/connhelper.go#L20-L35

**Review by suzuki-shunsuke — APPROVED** [review]

**Comment by suzuki-shunsuke** [comment]
Thanks a lot!

File: internal/provider/config.go (hunk 1 above)
Comment (innovate-invent) [reply internal/provider/config.go:122]: yep :+1:

**Pull request merged**